File: 11-container-with-most-water/container-with-most-water.py

class Solution:
    def maxArea(self, height):
        area_=0
        left=0
        right=len(height)-1
        while right> left:
            area=(right-left)* min(height[left],height[right])
            area_= max(area_,area)
            if height[right]> height[left]:
                left+=1
            else:
                right-=1
        return area_

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxLen = 0;
        while(left<right){
            int width = right-left;
            int h = Math.min(height[left],height[right]);
            maxLen = Math.max(maxLen,width*h);
            if(height[left]<height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxLen;
    }
}